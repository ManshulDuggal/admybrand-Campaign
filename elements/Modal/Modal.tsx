import React from "react";
import localStyles from "./styles.module.scss";
interface Props {
  handler(arg: boolean): void;
}
const Modal: React.FC<Props> = ({ handler, children }) => {
    const toggler = (e)=>{
        if (e.target === e.currentTarget) {
            handler(false);
          }
    }
  return (
    <div
      className={localStyles.wrapper}
      onClick={(e)=>toggler(e)}
    >
      {children}
    </div>
  );
};
export default Modal;
