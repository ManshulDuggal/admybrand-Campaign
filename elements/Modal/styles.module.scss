.wrapper{
    @apply z-50 fixed top-0 left-0 right-0 h-full w-screen flex justify-center items-center bg-black bg-opacity-10;
}