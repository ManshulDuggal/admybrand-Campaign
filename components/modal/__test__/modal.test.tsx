import { render } from "@testing-library/react";
import React from "react";
import Modal from "../modal";

describe("render test", () => {
  it("initial test", () => {
    const component = render(<Modal />);
    expect(component).toMatchSnapshot();
  });
});
