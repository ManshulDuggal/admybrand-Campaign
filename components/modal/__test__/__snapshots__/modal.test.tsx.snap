// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render test initial test 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="undefined"
      >
        <div>
          <div>
            <div>
              Customized ADmyBRAND
            </div>
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 1024 1024"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M563.8 512l262.5-312.9c4.4-5.2.7-13.1-6.1-13.1h-79.8c-4.7 0-9.2 2.1-12.3 5.7L511.6 449.8 295.1 191.7c-3-3.6-7.5-5.7-12.3-5.7H203c-6.8 0-10.5 7.9-6.1 13.1L459.4 512 196.9 824.9A7.95 7.95 0 0 0 203 838h79.8c4.7 0 9.2-2.1 12.3-5.7l216.5-258.1 216.5 258.1c3 3.6 7.5 5.7 12.3 5.7h79.8c6.8 0 10.5-7.9 6.1-13.1L563.8 512z"
              />
            </svg>
          </div>
          <div>
            Suggested Settings
            <div>
              <div>
                <span>
                  India
                </span>
                <span>
                  Country
                </span>
              </div>
              <div>
                <span>
                  English
                </span>
                <span>
                  Language
                </span>
              </div>
              <div>
                <span>
                  Indian Rupee
                </span>
                <span>
                  Currency
                </span>
              </div>
            </div>
          </div>
          <div>
            <div>
              Country
            </div>
            <div>
              Language
            </div>
            <div>
              Currency
            </div>
          </div>
          <div>
            Choose a 
            country
          </div>
          <div>
            <div>
              <h1>
                India
              </h1>
            </div>
            <div>
              <h1>
                Singapore
              </h1>
            </div>
            <div>
              <h1>
                United Kingdom
              </h1>
            </div>
            <div>
              <h1>
                United States
              </h1>
            </div>
            <div>
              <h1>
                Indonesia
              </h1>
            </div>
            <div>
              <h1>
                Malaysia
              </h1>
            </div>
            <div>
              <h1>
                Nepal
              </h1>
            </div>
            <div>
              <h1>
                Thailand
              </h1>
            </div>
            <div>
              <h1>
                Vietnam
              </h1>
            </div>
            <div>
              <h1>
                Canada
              </h1>
            </div>
            <div>
              <h1>
                Philippines
              </h1>
            </div>
            <div>
              <h1>
                UAE
              </h1>
            </div>
            <div>
              <h1>
                Australia
              </h1>
            </div>
            <div>
              <h1>
                Bulgaria
              </h1>
            </div>
            <div>
              <h1>
                Japan
              </h1>
            </div>
            <div>
              <h1>
                Israel
              </h1>
            </div>
            <div>
              <h1>
                Germany
              </h1>
            </div>
            <div>
              <h1>
                Sweden
              </h1>
            </div>
            <div>
              <h1>
                South Korea
              </h1>
            </div>
            <div>
              <h1>
                Italy
              </h1>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="undefined"
    >
      <div>
        <div>
          <div>
            Customized ADmyBRAND
          </div>
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 1024 1024"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M563.8 512l262.5-312.9c4.4-5.2.7-13.1-6.1-13.1h-79.8c-4.7 0-9.2 2.1-12.3 5.7L511.6 449.8 295.1 191.7c-3-3.6-7.5-5.7-12.3-5.7H203c-6.8 0-10.5 7.9-6.1 13.1L459.4 512 196.9 824.9A7.95 7.95 0 0 0 203 838h79.8c4.7 0 9.2-2.1 12.3-5.7l216.5-258.1 216.5 258.1c3 3.6 7.5 5.7 12.3 5.7h79.8c6.8 0 10.5-7.9 6.1-13.1L563.8 512z"
            />
          </svg>
        </div>
        <div>
          Suggested Settings
          <div>
            <div>
              <span>
                India
              </span>
              <span>
                Country
              </span>
            </div>
            <div>
              <span>
                English
              </span>
              <span>
                Language
              </span>
            </div>
            <div>
              <span>
                Indian Rupee
              </span>
              <span>
                Currency
              </span>
            </div>
          </div>
        </div>
        <div>
          <div>
            Country
          </div>
          <div>
            Language
          </div>
          <div>
            Currency
          </div>
        </div>
        <div>
          Choose a 
          country
        </div>
        <div>
          <div>
            <h1>
              India
            </h1>
          </div>
          <div>
            <h1>
              Singapore
            </h1>
          </div>
          <div>
            <h1>
              United Kingdom
            </h1>
          </div>
          <div>
            <h1>
              United States
            </h1>
          </div>
          <div>
            <h1>
              Indonesia
            </h1>
          </div>
          <div>
            <h1>
              Malaysia
            </h1>
          </div>
          <div>
            <h1>
              Nepal
            </h1>
          </div>
          <div>
            <h1>
              Thailand
            </h1>
          </div>
          <div>
            <h1>
              Vietnam
            </h1>
          </div>
          <div>
            <h1>
              Canada
            </h1>
          </div>
          <div>
            <h1>
              Philippines
            </h1>
          </div>
          <div>
            <h1>
              UAE
            </h1>
          </div>
          <div>
            <h1>
              Australia
            </h1>
          </div>
          <div>
            <h1>
              Bulgaria
            </h1>
          </div>
          <div>
            <h1>
              Japan
            </h1>
          </div>
          <div>
            <h1>
              Israel
            </h1>
          </div>
          <div>
            <h1>
              Germany
            </h1>
          </div>
          <div>
            <h1>
              Sweden
            </h1>
          </div>
          <div>
            <h1>
              South Korea
            </h1>
          </div>
          <div>
            <h1>
              Italy
            </h1>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
